### PR TITLE
Add executor of commands, execlib, and command example client

### PR DESCRIPTION
**Pull request by eriknordmark**
@rvs If this executor works for you (see example calls in command.go), then I can add it.
Note that every command runs in the foreground with a timelimit.

The execlib.New and exevlib.Execute is used by the example command.go to show how to use this.

### CODE REVIEW

File: pkg/pillar/zedbox/zedbox.go [hunk 1]
```diff
@@ -39,12 +41,17 @@ func main() {
...
+		command.Run()
 	case "diag":
 		diag.Run(ps)
 	case "domainmgr":
 		domainmgr.Run(ps)
 	case "downloader":
 		downloader.Run(ps)
+	case "executor":
+		// XXX add ps
+		executor.Run()
```
Comment (deitch) [inline pkg/pillar/zedbox/zedbox.go]: Please pass it `PubSub.pubsub` to keep with what the others do.

File: pkg/pillar/zedbox/zedbox.go [hunk 2]
```diff
@@ -39,12 +41,17 @@ func main() {
 	switch basename {
 	case "client":
 		client.Run(ps)
+	case "command":
+		command.Run()
```
Comment (deitch) [inline pkg/pillar/zedbox/zedbox.go]: Please pass it `ps` to keep with what the others do.

**Comment by eriknordmark** [comment]
> The bigger issue is that I don't know what this is supposed to be for. Why is one container executing a command in another? Is that not a security risk? Assuming we wanted to do things in others, would we not just pass it via pubsub messages?

Intended use is to replace the chroot shell script wrapper @rvs has for xl, xentop, etc commands. Not any less secure than the chroot shell script.

File: pkg/pillar/zedbox/zedbox.go (hunk 1 above)
Comment (eriknordmark) [reply pkg/pillar/zedbox/zedbox.go]: Done.

File: pkg/pillar/zedbox/zedbox.go (hunk 2 above)
Comment (eriknordmark) [reply pkg/pillar/zedbox/zedbox.go]: Done

File: pkg/pillar/zedbox/zedbox.go (hunk 1 above)
Comment (deitch) [reply pkg/pillar/zedbox/zedbox.go]: Thanks

**Comment by deitch** [comment]
> Intended use is to replace the chroot shell script wrapper @rvs has for xl, xentop, etc commands. Not any less secure than the chroot shell script

The whole "one container exec-ing into another to execute things" is a bit odd. Given that we do it, this is a better solution, but is there a cleaner way overall to do this kind of thing? Or do we necessarily expect that one "uber-container" (like pillar) is going to have more rights than others?

**Comment by eriknordmark** [comment]
@rvs still waiting to find out whether this matches what you were looking for.

**Pull request merged**